operation/fhir-operation-validate/src/main/java/org/linuxforhealth/fhir/operation/validate/ValidateOperation.java

```diff
@@ -163,9 +198,72 @@ public List<Issue> validateDeleteResource(String type, String id, FHIROperationC
         FHIRPersistence persistence =
                 (FHIRPersistence) operationContext.getProperty(FHIROperationContext.PROPNAME_PERSISTENCE_IMPL);
         if (!persistence.isDeleteSupported()) {
-            warnings.add(FHIRUtil.buildOperationOutcomeIssue(IssueSeverity.WARNING, IssueType.NOT_SUPPORTED, "Resource deletion of type '"
-                    + type + "' with id '" + id + "' is not supported."));
+            warnings.add(FHIRUtil.buildOperationOutcomeIssue(IssueSeverity.ERROR, IssueType.NOT_SUPPORTED, "Resource deletion, of type '"
+                    + type + "', with id '" + id + "', is not supported."));
         }
         return warnings;
     }
+    
+    /**
+     * Method to build the OperationOutcome error response
+     * @param severity the issue severity
+     * @param issueType describes the type of issue
+     * @param msg describes the error message
+     * @param cause the throwable that causes this OperationOutcome/Exception
+     * @return FHIROperationException with issues
+     * @throws FHIROperationException
+     */
+    protected FHIROperationException buildExceptionWithIssue(IssueSeverity severity, IssueType issueType, String msg, Throwable cause) throws FHIROperationException {
+        OperationOutcome.Issue ooi = FHIRUtil.buildOperationOutcomeIssue(severity, issueType, msg);
+        return new FHIROperationException(msg, cause).withIssue(ooi);
+    }
+    
+    /**
+     * Method to validate the resource parameter. The below validations are performed by this method.
+     * 1. if $validate is invoked at instance level and mode = create, check if the resource already exists.
+     * 2. resource parameter must be present unless the mode is "delete"
```

Comment:
Its probably worth noting here that we also allow the resource parameter to be absent when the mode is "profile".
I opened the follow issue against the specification so that we could reference it in our comment:  https://jira.hl7.org/browse/FHIR-37998
```suggestion
     * 2. resource parameter must be present unless the mode is "delete" (or "profile" per https://jira.hl7.org/browse/FHIR-37998)
```

Reply:
This has been fixed as suggested.